src/jit/compiler.cpp

```diff
@@ -2594,9 +2602,7 @@ void Compiler::compSetProcessor()
             codeGen->getEmitter()->SetContainsAVX(false);
             codeGen->getEmitter()->SetContains256bitAVX(false);
         }
-        else if (compSupports(InstructionSet_SSSE3) || compSupports(InstructionSet_SSE41) ||
-                 compSupports(InstructionSet_SSE42) || compSupports(InstructionSet_AES) ||
-                 compSupports(InstructionSet_PCLMULQDQ))
+        else if (compSupports(InstructionSet_SSSE3) || compSupports(InstructionSet_AES) || compSupports(InstructionSet_PCLMULQDQ))
```

Comment:
Now that the knobs are hierarchical, we only need to check `SSSE3` (which also controlls `SSE4.1` and `SSE4.2`), `AES`, and `PCLMULQDQ`.

Reply:
I'm also not sure we really need this setting. I don't think there is anything really special about the 4-byte instructions (other than them taking an additional byte) and there isn't any overlap with the 3-byte instructions (plus this only matters for the non-VEX case).

Reply:
> I'm also not sure we really need this setting. 

This is a workaround now, we would not need it if the emitter properly addresses 4-byte instructions independently from ISA.

Reply:
I don't think we really need it in general? Right now everything is controlled by:
```
bool emitter::Is4ByteSSE4Instruction(instruction ins)
{
    return UseSSE4() && IsSSE4Instruction(ins) && EncodedBySSE38orSSE3A(ins);
}
```

We could just get rid of the `UseSSE4()` check and us setting the value here and things would still work.
We could also just get rid of `IsSSE4Instruction(ins)` since (afair) just `EncodedBySSE38orSSE3A` is sufficient to determine if we need to use the 4-byte encoding.

Reply:
(the only real other requirement is that we return `false` if `UseVexEncoding()` returns `true`)